src/libnnstreamer-edge/nnstreamer-edge-internal.c

```diff
@@ -1357,6 +1453,22 @@ nns_edge_start (nns_edge_h edge_h)
 
   nns_edge_lock (eh);
 
+  if (NNS_EDGE_CONNECT_TYPE_CUSTOM == eh->connect_type) {
+    NnsEdgeCustomDef *custom_h = (NnsEdgeCustomDef *) eh->custom_h;
+    ret = custom_h->nns_edge_custom_start (eh->broker_h);
+    if (NNS_EDGE_ERROR_NONE != ret) {
+      nns_edge_loge ("Failed to start edge custom connection");
+    }
+    ret = custom_h->nns_edge_custom_set_event_cb (eh->broker_h, eh->event_cb,
+        eh->user_data);
+    if (NNS_EDGE_ERROR_NONE != ret) {
+      nns_edge_loge ("Failed to set event callback to custom connection.");
+      goto done;
+    }
+    ret = _nns_edge_create_send_thread (eh);
+    goto done;
+  }
+
```

Comment:
If connect_type is NNS_EDGE_CONNECT_TYPE_CUSTOM, you have written the start of nnstreamer edge well. :100: 
I think it would be good to make this part into a function. What do you think about it? :smiley_cat: 
```
 if (NNS_EDGE_CONNECT_TYPE_CUSTOM == eh->connect_type) {
    ret = start_custom_connection (eh);
    if (NNS_EDGE_ERROR_NONE != ret) {
      goto done;
    }
  }
```  
```
static int
start_custom_connection (nns_edge_handle_s * eh)
{
  int ret = NNS_EDGE_ERROR_NONE;
  NnsEdgeCustomDef *custom_h = (NnsEdgeCustomDef *) eh->custom_h;

  ret = custom_h->nns_edge_custom_start (eh->broker_h);
  if (NNS_EDGE_ERROR_NONE !=e ("Failed to start edge custom connection");
  }

  ret = custom_h->nns_edge_custom_set_event_cb (eh->broker_h, eh->event_cb,
      eh->user_data);
  if (NNS_EDGE_ERROR_NONE != ret) {
    nns_edge_loge ("Failed to set event callback to custom connection.");
    return ret;
  }

  ret = _nns_edge_create_send_thread (eh);

  return ret;
}
```If you have time, it would be easier to understand the code if you extract NNS_EDGE_NODE_TYPE_QUERY_SERVER and NNS_EDGE_CONNECT_TYPE_AITT, which are in if statements, into functions. :+1:

Reply:
:pray: Thank you. Have a nice day